### ISSUE TITLE
ActionView is still not independent of ActionPack

### ISSUE THREAD

**Issue by amatsuda** [rails/rails#17030]
actionview.gemspec [defines actionpack as a development_dependency](https://github.com/rails/rails/blob/13e1348cbad06c8dd20f03dc0cba334e591e5851/actionview/actionview.gemspec#L29) which of course means that ActionView is not referencing ActionPack in runtime, but [in fact ActionView references](https://github.com/search?utf8=%E2%9C%93&q=%22Mime%22+repo%3Arails%2Frails+path%3A%2Factionview%2Flib%2Faction_view&type=Code&ref=advsearch&l=) `::Mime` module which is defined [here in actionpack/lib/action_dispatch](https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/http/mime_type.rb#L6).

I'm not sure if there's any other dependency, but as for this Mime thing, I think we have three options:

1. Move Mime module to ActiveSupport
1. Let actionview depend on actionpack
1. Extract Mime into a dependent gem

WDYT?

**Comment by designcrumble** [comment]
I have just started using it and I have the following error while starting the server
![image](https://cloud.githubusercontent.com/assets/10060986/9087357/c484702c-3ba7-11e5-8d3f-994470590dd6.png)

**Comment by maclover7** [comment]
I think this can be closed -- ActionView is only defining ActionPack as a **development** dependency, meaning that it isn't required for "runtime" usage. Thus, this means that ActionView and ActionPack are decoupled from each other.

**Comment by claudiob** [comment]
@maclover7 Even with ActionView 5.0.0.beta1 I still get the same error detailed in the [original comment](https://github.com/rails/rails/issues/14686#issue-31250039):

```ruby
require 'action_view'
ActionView::version
# => Gem::Version.new("5.0.0.beta1")
paths = ActionView::PathSet.new(["test"])
lookup_context = ActionView::LookupContext.new(paths)
renderer = ActionView::Renderer.new(lookup_context)
view_context = ActionView::Base.new(renderer)
renderer.render(view_context, template: 'foo')
# NameError: uninitialized constant ActionView::LookupContext::DetailsKey::Mime
# from /Users/claudiob/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/actionview-5.0.0.beta1/lib/action_view/lookup_context.rb:70:in `get'
```

**Comment by maclover7** [comment]
Opened a PR!

**Comment by maclover7** [comment]
Solved via #23085.

**Comment by claudiob** [comment]
:boom: :boom: :boom: :boom: :boom: 
:clap: :clap: :clap: :clap: :clap: 
:+1: :+1: :+1: :+1: :+1:
